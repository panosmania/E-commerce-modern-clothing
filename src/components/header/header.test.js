import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

import { Header } from "./header.component";
import CartDropdown from "../cart-dropdown/cart-dropdown.component";

describe("Header component", () => {
  let wrapper;
  let mockSignOutStart;

  beforeEach(() => {
    mockSignOutStart = jest.fn();

    const mockProps = {
      hidden: true,
      currentUser: {
        uid: "123",
      },
      signOutStart: mockSignOutStart,
    };

    wrapper = shallow(<Header {...mockProps} />);
  });

  it("should render Header component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe("if currentUser is present", () => {
    it("should render sign out link", () => {
      expect(wrapper.find("div").at(2).text()).toBe("SIGN OUT");
    });

    it("should call signOutStart method when link is clicked", () => {
      wrapper.find(".option").at(2).simulate("click");

      expect(mockSignOutStart).toHaveBeenCalled();
    });
  });

  describe("if currentUser is null", () => {
    it("should render sign in link", () => {
      const mockProps = {
        hidden: true,
        currentUser: null,
        signOutStart: mockSignOutStart,
      };

      const newWrapper = shallow(<Header {...mockProps} />);

      expect(newWrapper.find(".option").at(2).text()).toBe("SIGN IN");
    });
  });

  describe("if hidden is true", () => {
    it("should not render CartDropdown", () => {
      expect(wrapper.exists(CartDropdown)).toBe(false);
    });
  });

  describe("if currentUser is null", () => {
    it("should render CartDropdown", () => {
      const mockProps = {
        hidden: false,
        currentUser: null,
        signOutStart: mockSignOutStart,
      };

      const newWrapper = shallow(<Header {...mockProps} />);

      expect(newWrapper.exists(CartDropdown)).toBe(true);
    });
  });
});
