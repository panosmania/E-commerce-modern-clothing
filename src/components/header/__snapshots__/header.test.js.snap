// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
<div
  className="header"
>
  <Link
    className="logo-container"
    to="/"
  >
    <ForwardRef(SvgLogo)
      className="logo"
    />
  </Link>
  <div
    className="options"
  >
    <Link
      className="option"
      to="/shop"
    >
      SHOP
    </Link>
    <Link
      className="option"
      to="/shop"
    >
      CONTACT
    </Link>
    <div
      className="option"
      onClick={[MockFunction]}
    >
      SIGN OUT
    </div>
    <Connect(CartIcon) />
  </div>
</div>
`;
